src/store/DevfileRegistries/index.ts

```diff
@@ -150,25 +154,38 @@ export const reducer: Reducer<State> = (state: State | undefined, incomingAction
     case 'REQUEST_METADATA':
     case 'REQUEST_SCHEMA':
     case 'REQUEST_DEVFILE':
-      return Object.assign({}, state, {
+      return createState(state, {
```

Comment:
Why do you decide to remove creating a new object and change the state directly?

Reply:
`createState` is the helper function that creates a new state object.

Reply:
see https://github.com/che-incubator/che-dashboard-next/pull/66/files#diff-9c8dc185fd5a878fd172c423064851faR22